core/session.ts

```diff
@@ -506,6 +517,21 @@ export class Session {
       this.compileError(new Error(message), firstActionInCycle.fileName);
     });
   }
+
+  private checkRunCachingCorrectness(actionsWithOutput: IActionProto[]) {
+    actionsWithOutput.forEach(action => {
+      if (action.dependencies?.length > 0) {
+        return;
+      }
+      if ([dataform.ActionHermeticity.HERMETIC, dataform.ActionHermeticity.NON_HERMETIC].includes(action.hermeticity)) {
+        return;
+      }
+      this.compileError(
```

Comment:
Can we add a todo to link to docs in this error?

Reply:
I don't think it's a good idea to link to docs in compile errors... I think it'll be very brittle, we'll change URLs / move content around and forget to update links. (This is made even worse by the compiler being inside `core`, so there would be absolutely no way of updating links without pushing out a new version of `core`.)